sympy/utilities/lambdify.py

```diff
@@ -92,6 +95,7 @@
     "mpmath": (MPMATH, MPMATH_DEFAULT, MPMATH_TRANSLATIONS, ("from mpmath import *",)),
     "numpy": (NUMPY, NUMPY_DEFAULT, NUMPY_TRANSLATIONS, ("import numpy; from numpy import *; from numpy.linalg import *",)),
     "scipy": (SCIPY, SCIPY_DEFAULT, SCIPY_TRANSLATIONS, ("import numpy; import scipy; from scipy import *; from scipy.special import *",)),
+    "cupy": (CUPY, CUPY_DEFAULT, CUPY_TRANSLATIONS, ("import cupy; import scipy; from scipy import *; from scipy.special import *",)),
```

Comment:
Can cupy be used with scipy functions?
I have seen that cupy have its own scipy implementation under `cupyx.scipy`
https://docs.cupy.dev/en/stable/reference/scipy.html
And it can be changed to

Reply:
Hi, thanks for the comment :)

Currently not all scipy is supported, just `ndimage` and `sparse`
I think I need to get ride of this. Thanks for noticing it!

Reply:
It's probably better to not mix SciPy with CuPy. Even if `scipy` works, they will be run on the CPU, which could be surprising for users. I think we should try to only use GPU functions, and if they aren't implemented yet, just leave them out until upstream support improves.

Reply:
yes, you're right :), thanks for the feedback!

Reply:
I completely agree with @asmeurer that it would be surprising for users if it runs on CPU. Better to let it fail explicitly.
From what I can tell, the only change this PR needs is the highlighted line, which should become the following.
`"cupy": (CUPY, CUPY_DEFAULT, CUPY_TRANSLATIONS, ("import cupy;",)),`